crates/sui-framework/packages/sui-framework/sources/crypto/nitro_attestation.move

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module sui::nitro_attestation;
```

Comment:
perhaps let's do:
```
module sui::enclave;

public struct AttestationDocument { all fields except for the cert & cabundle };
// getters...

/// Parses and verifies a given Nitro attestation document.
entry fun load_nitro_attestation_document(attestation: &vector<u8>, clock: &Clock): AttestationDocument {...}

```
(entry allows us to change the interface later, including to public when we like)

Reply:
there are some nuances on how to do this, ill check with Tim on how to do this with Option.

Reply:
```
    pub module_id: String,
    pub timestamp: u64,
    pub digest: String,
```
why are these necessary? or you think we should expose it for convenience in the future?

Reply:
mainly for future compatibility

Reply:
revised!